rust_snuba/src/processors/generic_metrics.rs

```diff
@@ -287,7 +311,13 @@ pub fn process_counter_message(
     _metadata: KafkaMessageMetadata,
     config: &ProcessorConfig,
 ) -> anyhow::Result<InsertBatch> {
-    process_message::<CountersRawRow>(payload, config)
+    let payload_bytes = payload.payload().context("Expected payload")?;
+    let msg: PartialMetricsMessage = serde_json::from_slice(payload_bytes)?;
```

Comment:
I tried to do some research because I was curious but I couldn't find anything: how does this save compute cost?

Reply:
The idea is, if on a pipeline there are 1000 messages, lets say with an average length of 1 KB. All the pipelines today need to deserialize 1000 * 1KB = 1MB of data with full deserialization. If lets say out of those 1000 messages, a pipeline is only interested in 10 of those (because they belong to its type of processing), then in theory it only needs to deserialize 10 * 1KB = 10KB of its own data. Lets say the partial message is of size 10. And in addition it needs to deserialize 1000 messages * 10 size = 10KB of all data. So in total, the pipeline is deserializing 10KB + 10KB = 20KB of data. Deserializing 20KB should take considerably lesser CPU than deserializing 1MB of data.

Reply:
I will write a benchmark test to see if I can show the improvement

Reply:
I would also be very curious. I would assume that the whole JSON still needs to be lexed at least to make sure that the parser can properly skip over properties that it is not interested in.
The only thing you are potentially saving here is converting the unused fields into proper data types, potentially allocating and copying strings around, etc.

A benchmark would surely be nice here.

Reply:
>  I would assume that the whole JSON still needs to be lexed at least to make sure that the parser can properly skip over properties that it is not interested in.

This is my concern too. Although maybe it's capable of bailing early if it finds the keys it's looking for? Anyways, this should be proved with a benchmark.

Reply:
Documented the results here https://www.notion.so/sentry/Benchmarking-performance-of-doing-partial-deserialization-in-snuba-generic-metrics-consumers-96a42d0039c84d018ce4dbdcfc15392a. We can clearly see an improvement with running the benchmark.